csrc/runtime/fusion_kernel_runtime.cpp

```diff
@@ -216,9 +216,17 @@ void FusionKernelRuntime::deserialize(
       runtime_id_ == buffer->runtime_id(),
       "Expected FusionKernelRuntime runtime_id to match serde runtime_id.");
 
+  // find the flatbuffer with the same group_id for SegmentedGroup
+  auto get_buffer = [&](int64_t group_id) {
+    for (auto buffer : *buffer->executors()) {
+      if (buffer->group_id() == group_id) {
+        return buffer;
+      }
+    }
```

Comment:
Clang Tidy is complaining something isn't always returned. Need to fix, will do in the `executor_dispatch` branch.